README.md

```diff
@@ -1,80 +1,444 @@
+│   │   │
+│   │   └── strategies/        # Attack strategies
+│   │       ├── incalmo_strategy.py    # Abstract base strategy class
+│   │       ├── strategy_factory.py    # Strategy instantiation factory
+│   │       ├── strategy_registry.py   # Dynamic strategy discovery and registration
+│   │       │
+│   │       ├── llm/               # LLM-based strategies
+│   │       │   ├── llm_strategy.py            # Base LLM strategy
+│   │       │   ├── langchain_strategy.py      # LangChain-based strategy implementation
+│   │       │   ├── langchain_registry.py      # LangChain tool registration
+│   │       │   ├── llm_agent_registry.py      # LLM agent registration
+│   │       │   ├── llm_response.py            # LLM response parsing and validation
+│   │       │   │
+│   │       │   └── interfaces/        # LLM interfaces and prompts
+│   │       │       ├── llm_interface.py
+│   │       │       ├── langchain_interface.py
+│   │       │       ├── llm_agent_interface.py
+│   │       │       └── preprompts/    # System prompts and templates
+│   │       │
+│   │       ├── state_machine/     # Rule-based strategies
+│   │       │   ├── __init__.py
+│   │       │   ├── graph_search.py        # Base graph search strategy
+│   │       │   ├── bfs.py                 # Breadth-first search strategy
+│   │       │   ├── dfs.py                 # Depth-first search strategy
+│   │       │   ├── struts_strategy.py     # Apache Struts exploitation chain
+│   │       │   ├── equifax_test.py        # Equifax breach simulation
+│   │       │   ├── MHBench_equifax_test.py # MHBench evaluation harness
+│   │       │   ├── darkside.py            # DarkSide ransomware simulation
+│   │       │   └── debug.py               # Debug/testing strategy
+│   │       │
+│   │       ├── testers/           # Strategy testing utilities
+│   │       │
+│   │       └── util/              # Strategy utilities
+│   │           └── event_util.py  # Event processing helpers
+│   │
+│   ├── frontend/              # Web interface
+│   │   └── incalmo-ui/        # React-based UI
+│   │       ├── .gitignore
+│   │       ├── package.json       # Node.js dependencies
+│   │       ├── tsconfig.json      # TypeScript configuration
+│   │       ├── README.md
+│   │       ├── public/            # Static assets
+│   │       └── src/               # React components and application logic
+│   │
+│   └── models/                # Shared data models (Pydantic)
+│       ├── __init__.py
+│       ├── agent.py               # Agent representation 
+│       ├── command.py             # Command structure with status enum
+│       ├── command_result.py      # Command execution results
+│       ├── instruction.py         # Agent instruction format
+│       ├── llm_agent_action_data.py   # LLM action data transfer object
+│       └── logging_schema.py      # Structured logging schemas
+└── output/                    # Execution logs and results
+
+```
+## Usage

```

Comment:
This is more useful than the project structure so it should be above it in the README (its best to order information by importance for the reader)